### DISCUSSION TITLE
Why are ConcurrentExecutingDisallowed and PersistJobDataAfterExecuting private/internal

### DISCUSSION THREAD

**Discussion by da3dsoul** [quartznet/quartznet#1890] [General]
Why are ConcurrentExecutingDisallowed and PersistJobDataAfterExecuting on JobDetailImpl private/internal?
I'm making a custom job builder, and it seems odd.
JobTypeInformation also seems like it should be public instead of internal.

**Comment by drieseng** [comment]
@da3dsoul, I suppose this is to encourage usage of **JobBuilder**. If you want/need a custom job builder, you can have it wrap the built-in one.

**Reply by da3dsoul** [reply]
That is what my original plan was, but a lot of JobBuilder's properties are private or internal, as well, so I can't even extend it without duplicating and reimplementing base code, not to mention the mess of not being able to restrict base class functions. If the properties were at least protected, I could extend it.
I ended up using reflection, but it's far from ideal.

**Comment by drieseng** [comment]
I don't see why you'd have to resort to reflection. Can you share what you've ended up doing, or shall we consider this discussion closed?

**Reply by da3dsoul** [reply]
https://github.com/da3dsoul/Quartz.Extensions.JobBuilderFactory/blob/c047d4d7b356c95554db89a98a0bdc4869c7f787/QuartzJobFactory/JobBuilder.cs#L51

I basically just replicated the JobBuilder with generic type parameters for the JobType.
